api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java

```diff
@@ -124,43 +128,97 @@ public class ObsBasedDiagnosticReportService extends BaseFhirService<DiagnosticR
 	@Qualifier("adminService")
 	private AdministrationService adminService;
 	
+	@Autowired
+	private ObservationTranslator observationTranslator;
+	
+	@Autowired
+	private DiagnosticReportObsLabResultTranslatorImpl diagnosticReportObsLabResultTranslator;
+	
 	@Override
 	public DiagnosticReport create(@Nonnull DiagnosticReport diagnosticReport) {
 		try {
 			diagnosticReportRequestValidator.validate(diagnosticReport);
+
+			List<Obs> reportResults = diagnosticReport.getResult().stream().map(reference -> {
+				IBaseResource obsResource = reference.getResource();
+				if ((obsResource != null) && (obsResource instanceof Observation) ) {
+					return observationTranslator.toOpenmrsType((Observation) obsResource);
+				} else {
+					return null;
+				}
+			}).filter(Objects::nonNull).collect(Collectors.toList());
+			diagnosticReport.setResult(Collections.emptyList());
+
 			FhirDiagnosticReport fhirDiagnosticReport = obsBasedDiagnosticReportTranslator.toOpenmrsType(diagnosticReport);
-			diagnosticReportObsValidator.validate(fhirDiagnosticReport);
-			
+			Set<Obs> attachmentObs = fhirDiagnosticReport.getResults();
+
 			Order order = getOrder(diagnosticReport, fhirDiagnosticReport);
 			Encounter encounter = createNewEncounterForReport(fhirDiagnosticReport, order);
+
 			fhirDiagnosticReport.setEncounter(encounter);
-			Set<Obs> reportObs = createReportObs(fhirDiagnosticReport, order, encounter);
+			if(attachmentObs.isEmpty()) {
+				Obs obs = reportResults.get(0);
+				LabResult labResult = LabResult.builder().
+						labResultValue(obs.getValueNumeric().toString())
+						.concept(fhirDiagnosticReport.getCode())
+						.obsFactory(newObs(fhirDiagnosticReport.getSubject(), fhirDiagnosticReport.getIssued()))
+						.build();
+				fhirDiagnosticReport.setResults(
+						of(diagnosticReportObsLabResultTranslator.toOpenmrsType(labResult))
```

Comment:
better to write Stream.of(..) .. clearer

Reply:
changes have been made